pkg/sql/parser/sql.y

```diff
@@ -6708,6 +6725,10 @@ col_qualification_elem:
   {
     $$.val = &tree.ColumnDefault{Expr: $2.expr()}
   }
+| ON UPDATE b_expr
+  {
+    $$.val = &tree.ColumnOnUpdate{Expr: $3.expr()}
```

Comment:
are we dealing with constraint name? there's some code here that assumes we are.

Reply:
Yes; because of the way the column qualifications are parsed, both the anonymous syntax `ON UPDATE expr` and the named syntax `CONSTRAINT name ON UPDATE expr` are valid uses.

Reply:
You should add a test in the Parser for this

Reply:
Done